src/components/FilterSearch.tsx

```diff
@@ -51,6 +51,15 @@ export interface FilterSearchProps {
   placeholder?: string,
   /** Whether to trigger a search when an option is selected. Defaults to false. */
   searchOnSelect?: boolean,
+  /** A function which is called when a filter is selected. To use this, searchOnSelect must be false. */
+  onSelect?: (
+    currentFilter: StaticFilter | undefined,
```

Comment:
So currentFilter was updated to allow any StaticFilter because the selected field value filter could transformed into a compound, conjunction filter for the UCSD case, is that correct?
I think it's confusing that the currently filter is of type `StaticFilter` and the newFilter is `FieldValueStaticFilter`. I would expect the current and new filter type to match.